### DISCUSSION TITLE
Is it possible to replace gulpfile.js & use ES Modules together?

### DISCUSSION THREAD

**Discussion by latushko-igor** [gulpjs/gulp#2669] [Help]
Hi.
I am trying to replase `gulpfile.js` how it described in docs and use ES6 import/export together.

I put my `gulpfile.js` in `/gulpfile.js` directory and rename it to `index.js`. Then add `"type": "module"` in `package.json`. And when i am start `gulp` i am get an error `Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'e:\..\gulpfile.js`.

Ok, then i run `gulp --gulpfile './gulpfile.js/index.js'` and get another error `ReferenceError: exports is not defined` pointed to string `exports.default = taskName` in my `./gulpfile.js/index.js`.

Is it possible to use together ES Modules and placed gulpfile in separate directory?

**UPD**: A solution is `export default taskName;` instead of `exports.default = taskName`

**Comment by artbindu** [comment]
I'm trying to same thing.

**Reply by artbindu** [reply]
https://stackoverflow.com/questions/74735718/require-of-es-module-gulpfile-js-not-supported-n-gulp-node-js
